core/modules/node/node.theme.inc

```diff
@@ -191,9 +191,12 @@ function template_preprocess_node(&$variables) {
   // Make the field variables available with the appropriate language.
   field_attach_preprocess('node', $node, $variables['content'], $variables);
 
-  // Display post information only on certain node types.
+  // Display post information only on certain node types or if it is a block
+  // that is configured to display 'node_submitted' info.
   $node_type = node_type_get_type($node->type);
-  if ($node_type->settings['node_submitted']) {
```

Comment:
I think what we may need is to check if `$variables['display_submitted']` is already set here. And if so, just respect the current setting rather than pulling from the `$node_type->settings`. We may need to add `display_submitted` to `node_theme()` so that the variable is passed correctly.

Reply:
Seems that the main issue is that in `template_preprocess_node` it assumes that the node in the block is using a view_mode called "block":

```
$variables['block']     = $variables['view_mode'] == 'block';
```